src/main/scala/pl/project13/scala/sbt/JmhPlugin.scala

```diff
@@ -2,18 +2,37 @@ package pl.project13.scala.sbt
 
 import sbt._
 import sbt.Keys._
+import java.io.PrintWriter
 import org.openjdk.jmh.annotations.Benchmark
 import org.openjdk.jmh.generators.bytecode.JmhBytecodeGenerator
-import sbt.CommandStrings._
-import java.io.PrintWriter
 
 import scala.tools.nsc.util.ScalaClassLoader.URLClassLoader
 
-object SbtJmh extends Plugin {
+object JmhPlugin extends AutoPlugin {
+
+  object JmhKeys {
+    val Jmh = config("jmh") extend Compile
+
+    val generateJavaSources = taskKey[Seq[File]]("Generate benchmark JMH Java code")
+
+    val outputTarget = settingKey[File]("Directory where the bytecode to be consumed and generated sources should be written to (`target` or sometimes `target/scala-2.10`)")
+
+    val generatorType = settingKey[String]("Benchmark code generator type. Available: `default`, `reflection` or `asm`.")
+
+    val generateInstrumentedClasses = taskKey[Seq[File]]("Generate instrumented JMH code")
+  }
 
   import JmhKeys._
 
-  lazy val jmhSettings = Seq(
```

Comment:
I'd say it warrants bump of the first version (e.g. `0.2.0`) as API is slightly different now and requires changes to existing builds.